fails to build with gcc 8.0.1?

arachne-pnr-2ae513555303fadcd80c96fd5204f86a6e75bcde]$ make -Wall
g++ -Isrc -std=c++11 -MD -O2 -Wall -Wshadow -Wsign-compare -Werror   -c -o src/arachne-pnr.o src/arachne-pnr.cc
In file included from /usr/include/c++/8/set:60,
                 from src/util.hh:25,
                 from src/bstream.hh:19,
                 from src/vector.hh:19,
                 from src/netlist.hh:21,
                 from src/arachne-pnr.cc:16:
/usr/include/c++/8/bits/stl_tree.h: In instantiation of ‘class std::_Rb_tree<Port*, Port*, std::_Identity<Port*>, IdLess, std::allocator<Port*> >’:
/usr/include/c++/8/bits/stl_set.h:133:17:   required from ‘class std::set<Port*, IdLess>’
src/netlist.hh:158:28:   required from here
/usr/include/c++/8/bits/stl_tree.h:452:21: error: static assertion failed: comparison object must be invocable with two arguments of key type
       static_assert(__is_invocable<_Compare&, const _Key&, const _Key&>{},
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [<builtin>: src/arachne-pnr.o] Error 1
[filiperosset@raw arachne-pnr-2ae513555303fadcd80c96fd5204f86a6e75bcde]$ gcc -v
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,objc,obj-c++,fortran,ada,go,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 8.0.1 20180410 (Red Hat 8.0.1-0.21) (GCC)

--- a/src/netlist.hh
+++ b/src/netlist.hh
@@ -51,9 +51,9 @@ public:
 class IdLess
 {
 public:
-  bool operator()(const Identified *lhs, const Identified *rhs) const
+  template<typename T> bool operator()(const T *lhs, const T *rhs) const
   {
-    return lhs->id < rhs->id;
+    return static_cast<const Identified*>(lhs)->id < static_cast<const Identified*>(rhs)->id;
   }
 };
 

